### ISSUE TITLE
[Q] No metrics logged in wandb using sb3 

### ISSUE THREAD

**Issue by maba1337** [wandb/wandb#4406]
Hello everyone,

i am using sb3 for RL. I want to use Wandb, but it only logs the system metrics and the gradients, but not the training metrics. This happens even with the examples listed on the documentation. I dont know how to solve this. 

Can anyone help me please. 

Thank you

**Comment by thanos-wandb** [comment]
Hi @maba1337 thank you for writing in! Would it be please possible to post your `wandb.init()` and `WandbCallback()` calls here? I have just tried this W&B integration example, and it seems to log the training metrics but it doesn't automatically generate a chart. You can get the training metrics by running the `Weave` expression `runs.history and then change the Render type to plot as below:

<img width="1291" alt="weave" src="https://user-images.githubusercontent.com/108423944/197583358-406cc805-ebf2-474b-b927-6cbc61312fa7.png">

Would this work for you?

**Comment by maba1337** [comment]
Hi @thanos-wandb thank you for your answer. I tried the same W&B integration example, so we used the same Callback config and wandb.init()
I see that you get that tensorflow icon in the left bar. This is something i never have. It only logged the metrics when i start the script as admin on a pc i had the admin permission. Could it be that wandb cant find the directory witht the tensorboard logs to upload it, because it has no permission? I also had some symlink errors with the directory for the model_save and tensorboard log path.

**Comment by thanos-wandb** [comment]
Hi @maba1337 thanks for the additional information. May I please ask if you're logging in our public cloud instance, or do you have a self-hosted instance? would it be possible to share the `debug.log` and `debug-internal.log` files from this run directory? also can you please check under `wandb/run-*/files/` and see if you have a `tfevents` file stored there?

**Comment by maba1337** [comment]
Hi @thanos-wandb , I log directly to the cloud. But even the offline run + starting the sync afterwards doesnt resolve the problem, that the tfevents wont be synced. The tfevets file has been created. The debug-internal.log also doesnt list that the tfevents file has been uploaded. 
[debug.log](https://github.com/wandb/wandb/files/9896472/debug.log)
[debug-internal.log](https://github.com/wandb/wandb/files/9896473/debug-internal.log)

So my guess really are some missing admin permission to created the TMPDIR for the tensorboard sync.

**Comment by maba1337** [comment]
I now tested to create the symlink of the tfevents file to the wandb/files folder of the run. Now the i can look up the tfevents in wandb but i doesnt log the sb3 metric graphs.

**Comment by tragram** [comment]
I can confirm that this is a problem on Windows 10 when using the cloud version of W&B - not even the example script for sb3 integration works. If I manually set the `tensorboard_log` path in sb3 to be inside the wandb folder, I can get the TF tab to appear in W&B. However, the data is still not available in the Charts tab, not even hidden (i.e. cannot create charts that plot train/*).

**Comment by maba1337** [comment]
Yesterday i found out a workaround! Maybe not the best solution but i dont have more time left for trying.
In the wandb/sync.py a TMPDIR will be created for the tb_log file. So my guess was, that the tbevents_metrics will be send to wandb from this TMPDIR. I just set this TMPDIR to the normal tbevents directory and the metrics have been logged in. 
@tragram I also did your solution to upload the tbevents log to wandb. 

There are 2 main points i found out. Because of missing admin rights, windows cant create a symlink to symlink the tbevents to the run/files directory and windows cant create the tmpdir inside the tmpdir and doesnt create an error message or something like that.

**Comment by thanos-wandb** [comment]
Hi @maba1337 thanks a lot for posting this workaround, glad to hear this now works for you! 

Since there is no issue with the training and the tfevents directory is created, you could also [sync the directory afterwards](https://docs.wandb.ai/guides/integrations/tensorboard#syncing-previous-tensorboard-runs) as: `wandb sync tfevents_dir`. Would this work for you?

**Comment by GKrafft2** [comment]
Hey I am having a similar probem. Only system info gets logged.

I'm running on windows 11

Here are pieces of the code I'm using: 

run = wandb.init(
    project="legTest2",
    sync_tensorboard=True,
    monitor_gym=True,
    save_code=False
)
 
(...)

sac_config={"learning_rate":1e-4,
            "buffer_size":300000,
            "batch_size":256,
            "ent_coef":'auto', 
            "gamma":0.99, 
            "tau":0.005,
            "train_freq":1, 
            "gradient_steps":1,
            "learning_starts": 10000,
            "verbose":1, 
            "tensorboard_log":f"runs/{run.id}",
            "policy_kwargs": policy_kwargs,
            "seed":None, 
            "device": gpu_arg}

(...)

model.learn(total_timesteps=1000000, log_interval=1, callback=WandbCallback(model_save_path=f"models/{run.id}"))

Find here my different debug files:
[debug.log](https://github.com/wandb/wandb/files/10197739/debug.log)
[debug-internal.log](https://github.com/wandb/wandb/files/10197740/debug-internal.log)
[debug-cli.eliot.log](https://github.com/wandb/wandb/files/10197741/debug-cli.eliot.log)

Note that the tensorboard data is created, as I can inspect it using tensorboard directly. 

I didn't quite understand the workaround @maba1337 used, if there is a more official answer that would be great.

Does anyone have an answer to that ?

Thanks in advance

**Comment by maba1337** [comment]
Hi @GKrafft2 , is it possible to start your training script as admin? if yes, then do it and tensorboard should get logged in wandb.

**Comment by GKrafft2** [comment]
It worked !

Thanks @maba1337

**Comment by Ammar-Azman** [comment]
I am using Fast RCNN from LayoutParser (https://github.com/Layout-Parser/layout-model-training), but WandB did not logged the metrics to its graph. Only system info get log. Same case as this question. The log information is kept in `log.txt` as the output of the training. Can you help me?

**Comment by 19kdc3** [comment]
I was running into this same issue and noticed that (at least for my custom environment) you need to take at least ~100 total time steps for train to log. 

Any one else still running into this issue, try increasing the number of total time steps.

**Comment by Ammar-Azman** [comment]
@maba1337 , nope, I ran it on Google Colab and also tried using AWS Sagemaker. But it still does not showed any graph on WandB side.
